astropy/convolution/convolve.py

```diff
@@ -338,21 +327,21 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
             padded_array = np.pad(array_internal, pad_width=np_pad_width,
                                   mode=np_pad_mode)
 
-        _convolveNd_padded_boundary_c(result, padded_array,
-                  array_internal.ndim,
-                  np.array(array_shape, dtype=ctypes.c_size_t, order='C'),
+        _convolveNd_c(result, padded_array,
+                  padded_array.ndim,
```

Comment:
Might it better to map the variants to common vars and then hoist the call to ``_convolveNd_c`` to after the condition?

Reply:
Do you mean having a 4-way option (padded + interpolate, unpadded + interpolate, etc) rather than two booleans?

Reply:
I just meant that since there is no only one func there should be only one call.
```python

array_to_convolve = padded_array if padded else array_internal

convolveNd_c(result, array_to_convolve,
                  array_to_convolve.ndim,
                  np.array(array_to_convolve.shape, dtype=ctypes.c_size_t, order='C'),
                  kernel_internal,
                  np.array(kernel_shape, dtype=ctypes.c_size_t, order='C'),
                  nan_interpolate,
                  nan_interpolate, padded,
                  n_threads
```
                  )

Reply:
Ah right yes, I can do that.